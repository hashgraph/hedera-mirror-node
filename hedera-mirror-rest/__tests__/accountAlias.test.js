/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import _ from 'lodash';

import AccountAlias from '../accountAlias';
import {getAllAccountAliases, invalidBase32Strs} from './testutils';

describe('AccountAlias', () => {
  describe('fromString', () => {
    describe('valid', () => {
      const testSpecs = [
        {
          input: 'AABBCC22',
          expected: new AccountAlias(null, null, 'AABBCC22'),
        },
        {
          input: '0.AABBCC22',
          expected: new AccountAlias(null, '0', 'AABBCC22'),
        },
        {
          input: '0.1.AABBCC22',
          expected: new AccountAlias('0', '1', 'AABBCC22'),
        },
        {
          input: '99999.99999.AABBCC22',
          expected: new AccountAlias('99999', '99999', 'AABBCC22'),
        },
      ];

      testSpecs.forEach((spec) => {
        test(spec.input, () => {
          expect(AccountAlias.fromString(spec.input)).toEqual(spec.expected);
        });
      });
    });

    describe('invalid', () => {
      const inputs = _.flattenDeep([
        null,
        undefined,
        invalidBase32Strs.map((alias) => getAllAccountAliases(alias)),
        '100000.100000.AABBCC22',
        '0.0.0.AABBCC22',
      ]);

      inputs.forEach((input) => {
        test(`${input}`, () => {
          expect(() => AccountAlias.fromString(input)).toThrowErrorMatchingSnapshot();
        });
      });
    });
  });

  describe('toString', () => {
    test('only alias', () => {
      const accountAlias = new AccountAlias(null, null, 'AABBCC22');
      expect(accountAlias.toString()).toBe('AABBCC22');
    });
    test('realm and alias', () => {
      const accountAlias = new AccountAlias(null, '0', 'AABBCC22');
      expect(accountAlias.toString()).toBe('0.AABBCC22');
    });
    test('shard, realm and alias', () => {
      const accountAlias = new AccountAlias('0', '1', 'AABBCC22');
      expect(accountAlias.toString()).toBe('0.1.AABBCC22');
    });
  });
});
