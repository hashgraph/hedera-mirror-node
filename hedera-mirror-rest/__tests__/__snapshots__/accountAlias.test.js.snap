// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountAlias fromString invalid  1`] = `"Invalid accountAlias string "`;

exports[`AccountAlias fromString invalid 0. 1`] = `"Invalid accountAlias string 0."`;

exports[`AccountAlias fromString invalid 0.0. 1`] = `"Invalid accountAlias string 0.0."`;

exports[`AccountAlias fromString invalid 0.0.0.AABBCC22 1`] = `"Invalid accountAlias string 0.0.0.AABBCC22"`;

exports[`AccountAlias fromString invalid 0.0.00 1`] = `"Invalid accountAlias string 0.0.00"`;

exports[`AccountAlias fromString invalid 0.0.11 1`] = `"Invalid accountAlias string 0.0.11"`;

exports[`AccountAlias fromString invalid 0.0.88 1`] = `"Invalid accountAlias string 0.0.88"`;

exports[`AccountAlias fromString invalid 0.0.99 1`] = `"Invalid accountAlias string 0.0.99"`;

exports[`AccountAlias fromString invalid 0.0.A 1`] = `"Invalid accountAlias string 0.0.A"`;

exports[`AccountAlias fromString invalid 0.0.AA====== 1`] = `"Invalid accountAlias string 0.0.AA======"`;

exports[`AccountAlias fromString invalid 0.0.AAA 1`] = `"Invalid accountAlias string 0.0.AAA"`;

exports[`AccountAlias fromString invalid 0.0.AAAAAA 1`] = `"Invalid accountAlias string 0.0.AAAAAA"`;

exports[`AccountAlias fromString invalid 0.0.aa 1`] = `"Invalid accountAlias string 0.0.aa"`;

exports[`AccountAlias fromString invalid 0.00 1`] = `"Invalid accountAlias string 0.00"`;

exports[`AccountAlias fromString invalid 0.11 1`] = `"Invalid accountAlias string 0.11"`;

exports[`AccountAlias fromString invalid 0.88 1`] = `"Invalid accountAlias string 0.88"`;

exports[`AccountAlias fromString invalid 0.99 1`] = `"Invalid accountAlias string 0.99"`;

exports[`AccountAlias fromString invalid 0.A 1`] = `"Invalid accountAlias string 0.A"`;

exports[`AccountAlias fromString invalid 0.AA====== 1`] = `"Invalid accountAlias string 0.AA======"`;

exports[`AccountAlias fromString invalid 0.AAA 1`] = `"Invalid accountAlias string 0.AAA"`;

exports[`AccountAlias fromString invalid 0.AAAAAA 1`] = `"Invalid accountAlias string 0.AAAAAA"`;

exports[`AccountAlias fromString invalid 0.aa 1`] = `"Invalid accountAlias string 0.aa"`;

exports[`AccountAlias fromString invalid 00 1`] = `"Invalid accountAlias string 00"`;

exports[`AccountAlias fromString invalid 11 1`] = `"Invalid accountAlias string 11"`;

exports[`AccountAlias fromString invalid 88 1`] = `"Invalid accountAlias string 88"`;

exports[`AccountAlias fromString invalid 99 1`] = `"Invalid accountAlias string 99"`;

exports[`AccountAlias fromString invalid 100000.100000.AABBCC22 1`] = `"Invalid accountAlias string 100000.100000.AABBCC22"`;

exports[`AccountAlias fromString invalid A 1`] = `"Invalid accountAlias string A"`;

exports[`AccountAlias fromString invalid AA====== 1`] = `"Invalid accountAlias string AA======"`;

exports[`AccountAlias fromString invalid AAA 1`] = `"Invalid accountAlias string AAA"`;

exports[`AccountAlias fromString invalid AAAAAA 1`] = `"Invalid accountAlias string AAAAAA"`;

exports[`AccountAlias fromString invalid aa 1`] = `"Invalid accountAlias string aa"`;

exports[`AccountAlias fromString invalid null 1`] = `"Invalid accountAlias string null"`;

exports[`AccountAlias fromString invalid undefined 1`] = `"Invalid accountAlias string undefined"`;
